extendedcopy.go

```diff
@@ -177,9 +177,15 @@ func findRoots(ctx context.Context, storage content.ReadOnlyGraphStorage, node o
 }
 
 // FilterAnnotation will configure opts.FindPredecessors to filter the predecessors
-// whose annotation matches a given regex pattern. For performance consideration,
-// when using both FilterArtifactType and FilterAnnotation, it's recommended to call
-// FilterArtifactType first.
+// whose annotation matches a given regex pattern. For each predecessor,
+// - If key is available in the predecessor's annotation and regex is not nil,
+// return the predecessor if the annotation value matches the regex.
+// - If key is available in the predecessor's annotation and regex is nil,
+// return the predecessor.
+// - If key is unavailable in the predecessor's annotation, the predecessor
+// will not be returned.
```

Comment:
A predecessor is not filtered out if the key is in its annotation and matches the regex if present.